### DISCUSSION TITLE
How to get database column name?

### DISCUSSION THREAD

**Discussion by ifeltsweet** [mikro-orm/mikro-orm#2288] [Q&A] [answered]
I know this was asked before because I've seen the question somewhere in the issues, but for the life of me, I can't find it again.

I want to be able to call some function from MikroORM that would return me the database column name according to the current naming strategy and any `fieldName` overrides used in the `Property` decorator so that I can safely reference columns in the raw SQL queries.

Contrived example:

```
knex.raw(`SELECT ${em.getDbColumnName(MyEntity, 'myEntityField')} FROM ${em.getTableName(MyEntity)}`);
```

**Comment by B4nan** [comment, marked as answer]
You could use the metadata directly:

```ts
const meta = em.getMetadata().getMetadata(MyEntity.name);
const fieldName = meta.properties.myEntityField.fieldNames[0];
```

edit: since v5.7.12 you can do just `em.getMetadata(MyEntity)` or `orm.getMetadata(MyEntity)`

**Reply by alex996** [reply]
Looks like `fieldName` is now `fieldNames`. But is there a scenario where this array has more than one column name?

**Reply by B4nan** [reply]
Yes, composite keys.

**Comment by Heniker** [comment]
Could this be mentioned in the docs? Perhaps under "Using Query Builder"? This is not very obvious and there are use-cases when one needs to use raw queries.
